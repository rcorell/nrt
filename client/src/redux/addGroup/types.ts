import { ActionType } from 'typesafe-actions';

import * as addGroupActions from './actions';

export interface AddGroupState {
    description: string;
    name: string;
}

export type AddGroupActions = ActionType<typeof addGroupActions>;
