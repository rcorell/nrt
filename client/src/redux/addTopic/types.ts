import { ActionType } from 'typesafe-actions';

import * as addTopicActions from './actions';

export interface AddTopicState {
    topic: string;
}

export type AddTopicActions = ActionType<typeof addTopicActions>;
