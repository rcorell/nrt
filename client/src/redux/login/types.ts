import { ActionType } from 'typesafe-actions';

import * as loginActions from './actions';

export enum LoginAttemptStatus {
    FAILURE = 'FAILURE',
    IN_PROGRESS = 'IN_PROGRESS',
    NOT_ATTEMPTED = 'NOT_ATTEMPTED',
    SUCCESS = 'SUCCESS'
}

export interface LoginState {
    authenticated: boolean;
    loginAttemptStatus: LoginAttemptStatus;
}

export type LoginActions = ActionType<typeof loginActions>;
