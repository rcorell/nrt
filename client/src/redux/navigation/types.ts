export enum Path {
    ADD_TOPIC = '/add-topic',
    HOME = '/',
    LOGIN = '/login',
    OUT_OF_SERVICE = '/out-of-service',
    SIGNUP = '/signup',
    TOPICS = '/topics',
    WAITING = '/waiting'
}
