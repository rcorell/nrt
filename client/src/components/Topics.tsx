import { useQuery } from '@apollo/client';
import _ from 'lodash';
import MaterialTable from 'material-table';
import React from 'react';

import { FetchTopicsQuery } from 'src/api/__generated__/FetchTopicsQuery';
import { fetchTopicsQuery } from 'src/api/api';
import { LOADING_TEXT } from 'src/components/shared';
import { setBrowserTitle } from 'src/utils';

const COLUMN_DEFINITIONS = [
    { field: 'title', title: 'Title' },
    { field: 'description', title: 'Description' },
    { field: 'group', title: 'Group' }
];

export const Topics: React.FC = () => {
    setBrowserTitle('Topics');

    const { data, error, loading } = useQuery<FetchTopicsQuery>(fetchTopicsQuery, { fetchPolicy: 'network-only' });

    if (loading) {
        return <div>{LOADING_TEXT}</div>;
    }

    if (error) {
        return <div>Error: {JSON.stringify(error)}</div>;
    }

    const topics = _.cloneDeep(data!.topics).map((topic) => ({ ...topic, group: topic.group.name }));

    return (
        <MaterialTable
            title="Topics"
            columns={COLUMN_DEFINITIONS}
            data={topics}
            options={{
                pageSize: 20,
                pageSizeOptions: [20, 50, 100]
            }}
        />
    );
};
