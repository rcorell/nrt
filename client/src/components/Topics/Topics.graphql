query FetchTopics {
        topics {
            id
            title
            description
            group {
                id
                name
            }
        }
    }