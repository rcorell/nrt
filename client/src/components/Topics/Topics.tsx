import _ from 'lodash';
import MaterialTable from 'material-table';
import React from 'react';

import { LOADING_TEXT } from 'src/components/shared';
import { setBrowserTitle } from 'src/utils';
import { useTopics } from './Topics.hook';

const COLUMN_DEFINITIONS = [
    { field: 'title', title: 'Title' },
    { field: 'description', title: 'Description' },
    { field: 'group', title: 'Group' }
];

export const Topics: React.FC = () => {
    setBrowserTitle('Topics');

    const useTopicsHook = useTopics();

    if (useTopicsHook.loading) {
        return <div>{LOADING_TEXT}</div>;
    }

    if (useTopicsHook.error) {
        return <div>Error: {JSON.stringify(useTopicsHook.error)}</div>;
    }

    const topics = _.cloneDeep(useTopicsHook.topics!).map((topic) => ({ ...topic, group: topic.group.name }));

    return (
        <MaterialTable
            title="Topics"
            columns={COLUMN_DEFINITIONS}
            data={topics}
            options={{
                pageSize: 20,
                pageSizeOptions: [20, 50, 100]
            }}
        />
    );
};
