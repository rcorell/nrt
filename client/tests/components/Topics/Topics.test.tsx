import { FetchTopicsQuery } from 'src/api/__generated__/types';
import { Topics } from 'src/components/Topics/Topics';
import { createComponentMocks, testComponent } from 'tests/testHelpers';

const mocks = createComponentMocks<FetchTopicsQuery>({
    topics: [
        {
            description: 'd1',
            group: {
                id: '1',
                name: 'g1'
            },
            id: '1',
            title: 't1'
        }
    ]
});

jest.mock('src/components/Topics/Topics.hook', () => ({
    useTopics: () => mocks[mocks.mockState]
}));

testComponent(Topics, mocks);
