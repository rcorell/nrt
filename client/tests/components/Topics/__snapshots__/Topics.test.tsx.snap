// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No displayName failure 1`] = `
<div>
  <div>
    Error: 
    "Error"
  </div>
</div>
`;

exports[`No displayName loading 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`No displayName success 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    style="position: relative;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MTableToolbar-root-5 MuiToolbar-gutters"
    >
      <div
        class="MTableToolbar-title-9"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
          style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
        >
          Topics
        </h6>
      </div>
      <div
        class="MTableToolbar-spacer-7"
      />
      <div
        class="MuiFormControl-root MuiTextField-root MTableToolbar-searchField-10"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
              title="Search"
            >
              search
            </span>
          </div>
          <input
            aria-invalid="false"
            aria-label="Search"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
            placeholder="Search"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="Clear Search"
              class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <span
                  aria-hidden="true"
                  aria-label="clear"
                  class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
                >
                  clear
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="MTableToolbar-actions-8"
      >
        <div>
          <div>
            <span />
          </div>
        </div>
      </div>
    </div>
    <div
      class="Component-horizontalScrollContainer-12"
      style="overflow-x: auto; position: relative;"
    >
      <div>
        <div
          style="overflow-y: auto;"
        >
          <div>
            <table
              class="MuiTable-root"
              style="table-layout: auto;"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-alignLeft"
                    scope="col"
                    style="box-sizing: border-box;"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiTableSortLabel-root"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        aria-describedby="rbd-hidden-text-0-hidden-text-0"
                        data-rbd-drag-handle-context-id="0"
                        data-rbd-drag-handle-draggable-id="0"
                        data-rbd-draggable-context-id="0"
                        data-rbd-draggable-id="0"
                        draggable="false"
                        role="button"
                        tabindex="0"
                      >
                        Title
                      </div>
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      >
                        arrow_downward
                      </span>
                    </span>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-alignLeft"
                    scope="col"
                    style="box-sizing: border-box;"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiTableSortLabel-root"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        aria-describedby="rbd-hidden-text-0-hidden-text-0"
                        data-rbd-drag-handle-context-id="0"
                        data-rbd-drag-handle-draggable-id="1"
                        data-rbd-draggable-context-id="0"
                        data-rbd-draggable-id="1"
                        draggable="false"
                        role="button"
                        tabindex="0"
                      >
                        Description
                      </div>
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      >
                        arrow_downward
                      </span>
                    </span>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-alignLeft"
                    scope="col"
                    style="box-sizing: border-box;"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiTableSortLabel-root"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        aria-describedby="rbd-hidden-text-0-hidden-text-0"
                        data-rbd-drag-handle-context-id="0"
                        data-rbd-drag-handle-draggable-id="2"
                        data-rbd-draggable-context-id="0"
                        data-rbd-draggable-id="2"
                        draggable="false"
                        role="button"
                        tabindex="0"
                      >
                        Group
                      </div>
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      >
                        arrow_downward
                      </span>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                  index="0"
                  level="0"
                  path="0"
                  style="transition: all ease 300ms;"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                    style="box-sizing: border-box; font-family: inherit; font-weight: inherit;"
                    value="t1"
                  >
                    t1
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                    style="box-sizing: border-box; font-family: inherit; font-weight: inherit;"
                    value="d1"
                  >
                    d1
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                    style="box-sizing: border-box; font-family: inherit; font-weight: inherit;"
                    value="g1"
                  >
                    g1
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 49px;"
                />
                <tr
                  class="MuiTableRow-root"
                  style="height: 1px;"
                />
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <table
      class="MuiTable-root"
    >
      <tfoot
        class="MuiTableFooter-root"
        style="display: grid;"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-footer"
        >
          <td
            class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root Component-paginationRoot-1"
            colspan="3"
            icons="[object Object]"
            style="float: right; overflow-x: auto;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-2 MuiToolbar-gutters"
            >
              <div
                class="MuiTablePagination-spacer"
              />
              <p
                class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-3 MuiTypography-body2 MuiTypography-colorInherit"
                id="mui-92599"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot Component-paginationSelectRoot-4"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="mui-92599 mui-29335"
                  class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                  id="mui-29335"
                  role="button"
                  tabindex="0"
                >
                  <div
                    style="padding: 0px 5px;"
                  >
                    20 rows 
                  </div>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value="20"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-3 MuiTypography-body2 MuiTypography-colorInherit"
              >
                1-1 of 1
              </p>
              <div
                class="MTablePaginationInner-root-14"
              >
                <span
                  class=""
                  title="First Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        first_page
                      </span>
                    </span>
                  </button>
                </span>
                <span
                  class=""
                  title="Previous Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        chevron_left
                      </span>
                    </span>
                  </button>
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                  style="flex: 1; text-align: center; align-self: center; flex-basis: inherit;"
                >
                  1-1 of 1
                </span>
                <span
                  class=""
                  title="Next Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        chevron_right
                      </span>
                    </span>
                  </button>
                </span>
                <span
                  class=""
                  title="Last Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        last_page
                      </span>
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
